site/zenodo_rdm/legacy/services.py

```diff
@@ -331,18 +332,33 @@ def get_record_by_bucket_id(self, bucket_id):
 
     def get_file_key_by_id(self, pid_value, file_id):
         """Get the associated record file key by its ID."""
-        RDMDraft = self.record_cls.model_cls
-        key = (
-            db.session.query(ObjectVersion.key)
-            .join(FileInstance, ObjectVersion.file_id == FileInstance.id)
-            .join(RDMDraft, ObjectVersion.bucket_id == RDMDraft.bucket_id)
-            .join(PersistentIdentifier, PersistentIdentifier.object_uuid == RDMDraft.id)
-            .filter(
-                FileInstance.id == file_id,
-                PersistentIdentifier.pid_type == "recid",
-                PersistentIdentifier.pid_value == pid_value,
-                PersistentIdentifier.object_type == "rec",
+
+        def query_file_id_by_model_cls(model_cls):
```

Comment:
I think this is used only for drafts...I am not sure if having this fallback makes sense